File: randomizeSet.cpp
#include<iostream>
#include<vector>
#include<map>
#include<ctime>
#include<cstdlib>
#include<time.h>
using namespace std;

class RandomizedSet {
public:
    RandomizedSet() {
        srand((unsigned int) time(NULL));
    }
    
    bool insert(int val) {
        if (hash.find(val)!=hash.end())
        {
            return false;
        }else{
            int n=vec.size();
            vec.push_back(val);
            hash[val]=n;
            return true;
        }
    }
    
    bool remove(int val) {
        if (hash.find(val)!=hash.end())
        {
            int temp=*(vec.end()-1);
            vec[hash[val]]=temp;
            hash[temp]=hash[val];
            vec.pop_back();
            hash.erase(val);
            return true;
        }else{
            return false;
        }
    }
    
    int getRandom() {
        int n=vec.size();
        if (n==1)
        {
            return vec[0];
        }else
        {
            int randomNum=rand()%n;
            return vec[randomNum];
        }
    }

private:
    vector<int> vec;
    map<int,int> hash;
};

int main(){
    RandomizedSet* obj = new RandomizedSet();
    cout<<obj->insert(0)<<endl;
    cout<<obj->insert(1)<<endl;
    cout<<obj->remove(0)<<endl;
    cout<<obj->insert(2)<<endl;
    cout<<obj->remove(1)<<endl;
    cout<<obj->getRandom()<<endl;
    return 0;
}
/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet* obj = new RandomizedSet();
 * bool param_1 = obj->insert(val);
 * bool param_2 = obj->remove(val);
 * int param_3 = obj->getRandom();
 */

File: randomizeSet.py

import random
class RandomizedSet:
    def __init__(self):
        self.vec=[]
        self.hash={}
    
    def insert(self, val: int) -> bool:
        if val in self.vec:
            return False
        n=len(self.vec)
        self.hash[val]=n
        self.vec.append(val)
        return True

    def remove(self, val: int) -> bool:
        if val not in self.vec:
            return False
        self.vec[self.hash[val]]=self.vec[-1]
        self.hash[self.vec[-1]]=self.hash[val]
        self.vec.pop()
        del self.hash[val]
        return True
    
    def getRandom(self) -> int:
        if len(self.vec)==1:
            return self.vec[0]
        else:
            return random.choice(self.vec)

if __name__=="__main__":
    obj = RandomizedSet()
    param_1 = obj.insert(1)
    param_1 = obj.insert(2)
    param_2 = obj.remove(1)
    param_3 = obj.getRandom()